mbuild/compound.py

```diff
@@ -312,8 +323,205 @@ def particles_by_name(self, name):
             if particle.name == name:
                 yield particle
 
+    @property
+    def contains_rigid(self):
+        """Returns True if the Compound contains rigid bodies
+
+        If the Compound contains any particle with a rigid_id != None 
+        then contains_rigid will return True. If the Compound has no
+        children (i.e. the Compound resides at the bottom of the containment
+        hierarchy) then contains_rigid will return False.
+
+        Returns
+        -------
+        bool
+            True if the Compound contains any particle with a rigid_id != None
+
+        """
+        if any(particle.rigid_id is not None for particle in self._particles()):
+            return True
+        else:
+            return False
+
+    @property
+    def max_rigid_id(self):
+        """Returns the maximum rigid body ID contained in the Compound.
+
+        This is usually used by compound.root to determine the maximum
+        rigid_id in the containment hierarchy.
+
+        Returns
+        -------
+        int or None
+            The maximum rigid body ID contained in the Compound. If no
+            rigid body IDs are found, None is returned
+
+        """
+        if self.contains_rigid:
+            return max([particle.rigid_id for particle in self.particles()
+                        if particle.rigid_id is not None])
```

Comment:
I think you can rewrite this to not traverse the full particles generator twice (since you're doing that in `self.contains_rigid`).

I think the following will work:

```python
try:
    return max([particle.rigid_id for particle in self.particles()
               if particle.rigid_id is not None])
except ValueError:
    return
```